### DISCUSSION TITLE
environmental variables and runtimeConfig

### DISCUSSION THREAD

**Discussion by R0N1n-dev** [nuxt/nuxt#10703] [Questions (Nuxt 2)]
How do i get my site to work in deployment and fetch data with my variables.  i have an api-key that needs to stay hidden but is needed in the link for fetching and  fetching is done on client side.
Docs r not helping
am lost at this point and just frustrated. All projects on hold till i sort this and i feel like m just stupid now.

**Comment by danielroe** [comment]
If you need to fetch on client side, you need the API key for fetching, and the API key needs to be hidden, then you should instead create an endpoint using `serverMiddleware`, and fetch on client side from that endpoint. Within the endpoint, you are on your own server and can have access to the API key.

I'm assuming you're using Nuxt 2 as you've created this discussion in nuxt/nuxt.js, but if on Nuxt 3, you can do the same with a server route.

**Comment by R0N1n-dev** [comment]
Thanks. This helped. Now for dynamic routes

**Comment by R0N1n-dev** [comment]
Thank you. Now to figure it out for dynamic links

**Comment by R0N1n-dev** [comment]
Thank you. Helped Alot

**Comment by R0N1n-dev** [comment]
Thank you. After going thru docs and ur explanation, it worked. Now to 
expand on it  to dynamic routes
